torch/onnx/_internal/torchscript.py

```diff
@@ -0,0 +1,426 @@
+"""Utilities for manipulating the torch.Graph object and the torchscript."""
+
+# TODO(justinchuby): Move more of the symbolic helper functions here and expose
+# them to the user.
+
+import dataclasses
+import numbers
+import re
+from typing import Any, Dict, Iterable, Iterator, List, Tuple, Union
+
+from typing_extensions import Protocol, runtime_checkable
+
+import torch
+from torch import _C
+from torch._C import _onnx as _C_onnx
+from torch.onnx import _deprecation
+from torch.onnx._globals import GLOBALS
+from torch.onnx._internal import _beartype
+
+_ATTR_PATTERN = re.compile("^(.+)_(([ifstgz])|(ty))$")
+
+
+@runtime_checkable
+class GraphLike(Protocol):
```

Comment:
```suggestion
class _GraphLike(Protocol):
```
If we don't want users to instantiate it, better to prepend a `_` to make it private?

Reply:
GraphLike is used in the symbolic function annotations, so it is exported outside of this module (public in that sense). Since torchscript is in _internal, the module itself is private and we don't give users any guarantee. In the future when we want to export some of the features to users we can come back and decide if parts of the module needs refactoring.